rfcs/0003-api-unification.md

```diff
@@ -0,0 +1,277 @@
+- Feature Name: API interface
+- Start Date: 2018-10-30)
+- RFC PR:
+- Salt Issue:
+
+# Summary
+[summary]: #summary
+
+Salt Module Interface (SMI) concept introduction for virtual modules.
+
+# Motivation
+[motivation]: #motivation
+
+Any salt module has couple of specific properties we dealing every day
+with:
+
+- Fixed set of functions
+- Known functions signatures
+- Known structure of return
+
+This is true but for virtual modules. The virtual module covering
+several "fixed" or "physical" modules and behaves like that would be
+one module. But differences between those physical modules on
+different platforms makes such virtual module a moving target and
+unpredictable.
+
+Virtual modules concept is missing crucial part in the design:
```

Comment:
Is your proposal to just restrict the usage of this to virtual modules or do you see a wider usage being adopted?

Reply:
I would not be very intrusive and have it for now only for virtual modules. But we should be able to always extend this.